core-bundle/src/Cache/EntityTagger.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Cache;
+
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\Persistence\Mapping\MappingException;
+
+class EntityTagger
+{
+    /**
+     * @var array<string, ClassMetadata>
+     */
+    private array $classMetadata = [];
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(EntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @param iterable|string|object|null $target
+     *
+     * @return array<int, string>
+     */
+    public function getTags($target): array
+    {
+        if (empty($target)) {
+            return [];
+        }
+
+        if (\is_string($target)) {
+            if (class_exists($target, false)) {
```

Comment:
As using classes with `FooBar::class` does not autoload the class I think this check could lead to weird bugs. How about instead checking if it is a valid PHP FQCN?

```suggestion
            if (preg_match('/^(?:[a-z_\x80-\xff][a-z0-9_\x80-\xff]*\\\\?)+(?<!\\\\)$/i', $target)) {
```

Reply:
Agreed, this is probably better suited!